### ISSUE TITLE
Add a post connection/verification landing page to IIWBook

### ISSUE THREAD

**Issue by swcurran** [bcgov/von#272]
The current IIWBook demo is a bit hard to understand when the connection to IIWBook is established. The web page just stays on the screen showing the QR code.  We'd like the events of establishing the connection and completing the verification to trigger updates to the IIWBook pages.  Here's a proposal, but if there are easier things that can be do to achieve the goal - showing something happened - then feel free to adjust.  This should be done quick and dirty as there is not a lot of value in this, so please take liberties in the implementation.  Peter Watkins @watkinspd would like a stable demo with this by next Wednesday.

Ideally, the flow would be something like this:

- When the QR code is read and the connection establishment process is happening/completes - go to a landing page with text on it to indicate that the connection is established, and waiting on the completion of the email proof.
- When the email proof completes - show the email address extracted from the proof and tell the user that their application to join IIWBook is being considered. If they are accepted, they will receive a notification in their mobile app of an offer to receive an IIWBook credential.
- If the page is still open when the backend process is completed, update the page to indicate that the credential offer has been sent.

Let me know if you want a first cut of the text, but the preference would be for you to do a cut of the text and when we see it, we'll collaborate on the exact wording presented at each stage.

The only three true requirements:

- At some point after the QR code is no longer needed, roll over to a new page with information on it.
- On the new page display the email address from the email verification service credential (once it's available)
- Display static text about what's happening/what's going to happen in the demo behind the scenes so the user knows what's going on.

FYI @jljordan42 

@nrempel - we're suggesting Andrew do this, but are familiar with this code such that it might be easier for you to do it, noting the deadline?

**Comment by nrempel** [comment]
This should be doable. I can take this one and work on it this weekend.

The approach I suggest (and will look into further):

- Create a javascript snippet that is loaded for every page that needs to listen for changes and poll an endpoint every second for state changes for this "session"
- When a change is discovered, javascript will trigger location change to load new page
- Continue for each transition
- The agent will listen to webhooks for relevant changes and update the state in the database for the polling endpoint to serve out

Websockets would be ideal here, but polling will be faster and should be fine. Django also doesn't play well with websockets out of the box so we avoid that trouble.

**Comment by swcurran** [comment]
Awesome - let me know when you have things to show so we can go over the text.

Thanks!

**Comment by nrempel** [comment]
I've updated https://iiwbook.vonx.io with a first cut of this flow. You should be able to walk through the process now and test it.

https://email-verification.vonx.io does not  seem to be  working  right now so I have been unable to test my work past the point  of  presenting the  verified  email proof. 

If  you have an old verified  email credential, you should be able  to walk through  the whole process.

**Comment by swcurran** [comment]
Looks great!  But I ran into a snag with it.  The first two steps worked
with the feedback, but when I completed the Proof Request/Proof submission,
the screen did not update.  Can you take a look?

I've submitted the proof and my application for inclusion into IIWBook has
gone to the backend and this is what I see.  And now I've completed the
backend process, received my Credential Offer, accepted that and received
the credential. And still the same screen is showing.

[image: image.png]
